server/src/main/java/org/opensearch/action/DocWriteRequest.java

```diff
@@ -248,6 +249,22 @@ static DocWriteRequest<?> readDocumentRequest(@Nullable ShardId shardId, StreamI
         return docWriteRequest;
     }
 
+    /**
+     * Validates whether the doc id length is under the limit.
+     * @param id DocId to verify
+     * @param validationException containing all the validation errors.
+     * @return validationException
+     */
+    static ActionRequestValidationException validateDocIdLength(String id, ActionRequestValidationException validationException) {
+        if (id != null && id.getBytes(StandardCharsets.UTF_8).length > 512) {
```

Comment:
I am wondering if we could use Apache Lucene's  `UnicodeUtil::calcUTF16toUTF8Length` without the need to allocate the byte[] for each string?